docs/clean/developer_guide/stages/steps.rst

```diff
@@ -0,0 +1,197 @@
+        CONFIG_REGISTER = '0x2102'
+        CURRENT_RUNNING_IMAGE = False
+
+        # ==============================
+        # Execution order of Stage steps
+        # ==============================
+        exec_order = [
+
+        ]
+
+        def delete_boot_variable(self, steps, device):
+
+            with steps.start("Delete any configure boot variables") as step:
+
+                try:
+                    device.configure("no boot system")
+                except Exception as e:
+                    step.failed("Failed to delete configured boot variables",
+                                from_exception=e)
+
+                step.passed("Successfully deleted configured boot variables")
+
+Passing Step Arguments
+----------------------
+
+To pass arguments to a clean step, add the arguments to the method and if there's a default, set it equal to that. Here's an example were we pass `images`, `timeout`, and `current_running_image`. You can see we provided the defaults by using:
+
+`timeout=TIMEOUT` `current_running_image=CURRENT_RUNNING_IMAGE`
+
+During runtime these values will be overwritten with user provided (if provided). You can then use these arguments like you would in any other python method.
+
+.. code-block:: python
+    :linenos:
+    :emphasize-lines: 19, 21, 42-43
+
+    from genie.metaparser.util.schemaengine import Optional
+    from genie.libs.clean import BaseStage
+
+    class ChangeBootVariable(BaseStage):
+
+        # ============
+        # Stage Schema
+        # ============
+        schema = {
+            Optional('images'): list,
+            Optional('timeout'): int,
+            Optional('config_register'): str,
+            Optional('current_running_image'): bool,
+        }
+
+        # =================
+        # Argument Defaults
+        # =================
+        TIMEOUT = 300
+        CONFIG_REGISTER = '0x2102'
+        CURRENT_RUNNING_IMAGE = False
+
+        # ==============================
+        # Execution order of Stage steps
+        # ==============================
+        exec_order = [
+
+        ]
+
+        def delete_boot_variable(self, steps, device):
+
+            with steps.start("Delete any configure boot variables") as step:
+
+                try:
+                    device.configure("no boot system")
+                except Exception as e:
+                    step.failed("Failed to delete configured boot variables",
+                                from_exception=e)
+
+                step.passed("Successfully deleted configured boot variables")
+
+        def configure_boot_variable(self, steps, device, images, timeout=TIMEOUT,
+                                    current_running_image=CURRENT_RUNNING_IMAGE):
+
+            with steps.start("Set boot variable to images provided for {}".format(
+                    device.name)) as step:
+
+                if current_running_image:
+                    log.info("Retrieving and using the running image due to "
```

Comment:
from here, user might copy and paste. I understand not main point. but might be better to have `import logging` and another line which define `log` at top.